server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/aws/AwsS3ObjectStorage.java

```diff
@@ -69,22 +72,37 @@
 
     public static Optional<PutBlobFunction> putBlob(BlobId.Factory blobIdFactory, ContainerName containerName, AwsS3AuthConfiguration configuration) {
         return Optional.of((blob) -> {
+            File file = null;
             try {
-                PutObjectRequest request = new PutObjectRequest(containerName.value(),
-                    blob.getMetadata().getName(),
-                    blob.getPayload().openStream(),
-                    new ObjectMetadata());
-
-                return blobIdFactory.from(getTransferManager(configuration)
-                    .upload(request)
-                    .waitForUploadResult()
-                    .getETag());
-            } catch (AmazonClientException | InterruptedException | IOException e) {
+                file = File.createTempFile(UUID.randomUUID().toString(), ".tmp");
+                FileUtils.copyToFile(blob.getPayload().openStream(), file);
+
+                return put(blobIdFactory, containerName, configuration, blob, file);
+            } catch (IOException e) {
                 throw new RuntimeException(e);
+            } finally {
+                if (file != null) {
+                    FileUtils.deleteQuietly(file);
+                }
             }
         });
     }
 
+    private static BlobId put(BlobId.Factory blobIdFactory, ContainerName containerName, AwsS3AuthConfiguration configuration, Blob blob, File file) {
```

Comment:
maybe you don't need to pass the blob as parameter now,  the name may be enough.

Reply:
I need it for the `name` & I'm not really happy of giving a `String` as a parameter.